src/lexer/lex.c

```diff
@@ -24,3 +24,110 @@ int unlex(Lexer *l, Token *t) {
     l->unlexed_count++;
     return 0;
 }
+
+TokenType ttype_one_char(char c) {
+    switch (c) {
+    case '(':
+        return TT_OPAREN; // (
+    case ')':
+        return TT_CPAREN; // )
+    case '{':
+        return TT_OBRACE; // {
+    case '}':
+        return TT_CBRACE; // }
+    case '[':
+        return TT_OBRACKET; // [
+    case ']':
+        return TT_CBRACKET; // ]
+    case ';':
+        return TT_SEMI; // ;
+    }
+
+    return TT_NO_TOKEN;
+}
+
+TokenType ttype_many_chars(const char *contents) {
+	// TODO: Handle operations
+
+    // Includes only numbers
+    int all_numeric = 1;
+
+    int count_fs = 0;
+    int count_us = 0;
+
+    int i;
+
+    if (contents == NULL) {
+        return TT_NO_TOKEN;
+    }
+
+    // Does it start with a negative sign
+	if (contents[0] == '-') {
+		return TT_IDENTIFIER;
+	}
+
+    // Loop through each character
+    for (i = 0; contents[i] != '\0'; i++) {
+        char c = contents[i];
+
+        // If it has a period, it's a float
+        if (c == '.') {
+            return TT_LITERAL;
+        }
+
+        // Count negative signs
+        if (c == '-') {
```

Comment:
In C, the expression `-2` is literally counted as `-` and then `2`, as two separate tokens. In the exact same way as you can say `int positive = 2, negative = -positive;` or whatever.